cpp/density/SphereVoxelization.cc

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2010-2019 The Regents of the University of Michigan
+// This file is from the freud project, released under the BSD 3-Clause License.
+
+#include <cmath>
+#include <stdexcept>
+#include <tbb/tbb.h>
+
+#include "SphereVoxelization.h"
+
+/*! \file SphereVoxelization.cc
+    \brief Routines for computing voxelized densities from spheres centered at points.
+*/
+
+namespace freud { namespace density {
+
+SphereVoxelization::SphereVoxelization(vec3<unsigned int> width, float r_max)
+    : m_box(box::Box()), m_width(width), m_r_max(r_max)
+{
+    if (r_max <= 0.0f)
+        throw std::invalid_argument("Sphere Voxelization requires r_max to be positive.");
+}
+
+//! Get a reference to the last computed voxels
+const util::ManagedArray<unsigned int>& SphereVoxelization::getVoxels() const
+{
+    return m_voxels_array;
+}
+
+//! Get width.
+vec3<unsigned int> SphereVoxelization::getWidth()
+{
+    return m_width;
+}
+
+//! internal
+/*! \brief Function to compute the voxels array
+ */
+void SphereVoxelization::compute(const freud::locality::NeighborQuery* nq)
+{
+    auto box = nq->getBox();
+    auto n_points = nq->getNPoints();
+    m_box = box;
+
+    vec3<unsigned int> width(m_width);
```

Comment:
Could this variable be removed and replaced by `m_width` everywhere?

Reply:
I think so. It looks like later in the method we go back to using `m_width` anyway. Do you remember what the original intent of defining the `width` variable was?

Reply:
Looks like its from `GaussianDensity`, do you remember the original intent of defining things this way?

Reply:
It turns out it is necessary, I'm not exactly sure why, other than in the 2D case, `m_voxels_array` ends up being 3 dimensional instead of 2

Reply:
This currently isn't a great design. The `width` variable is a copy of `m_width` but with z=1 if a 2D box is provided. This allows users to specify an integer for all dimensions, but it makes the behavior a bit ambiguous. If the user inputs `width=100` in the constructor, the `width` property returns `[100, 100, 100]` regardless of what dimensionality the box has... I'm not sure what to do about this but we can raise an issue.

Reply:
See #610. Closing this conversation for now -- this can be resolved later in a way that is consistent for GaussianDensity and SphereVoxelization.